tests/test_downloaders.py

```diff
@@ -449,7 +463,17 @@ def check_and_download_url_side_effect(url, download_dir, file_prefix='', **kwar
                 self.assertEqual(download_manager.download_dataset(dataset, ''), dataset_file_name)
                 self.assertTrue(logs_cm.records[0].message.startswith('Failed to download dataset'))
 
-    def test_download_dataset_failure(self):
+    def test_download_dataset_having_local_link_fails_because_of_corrupted_uri(self):
+        """Test that `download_dataset` raises a DownloadError exception if the download failed"""
+        download_manager = downloaders.DownloadManager()
+        dataset = Dataset.objects.get(pk=2)
+        with mock.patch.object(downloaders.HTTPDownloader, 'check_and_download_url') as mock_dl_url:
+            mock_dl_url.side_effect = downloaders.DownloadError
+            with self.assertRaises(downloaders.DownloadError):
+                with self.assertLogs(downloaders.LOGGER, logging.WARNING):
+                    download_manager.download_dataset(dataset, '')
+
+    def test_download_dataset_without_local_link_fails_because_of_corrupted_uri(self):
```

Comment:
I don't get why you added that test. The point of this test was to check that errors happening during the call to `check_and_download_url()` are correctly propagated. It has nothing to do with local URIs.

If you want to check that we don't attempt to download local URIs, you should write a test dedicated to that.

Reply:
These two test are actually real cases that simulates the cases in reality. They are kind of integration tests that can we nominate them. It does matter to have all kind of things in the dataset_uri model at the same time and the errors are properly raised!

Reply:
Done in the next commit. Another test is added to the suite.